sdk/identity/azure-identity/src/main/java/com/azure/identity/WorkloadIdentityCredential.java

```diff
@@ -89,6 +92,13 @@ public class WorkloadIdentityCredential implements TokenCredential {
         ClientAssertionCredential tempClientAssertionCredential = null;
         String tempClientId = null;
 
+        if (identityClientOptions.isAzureTokenProxyEnabled()) {
+            ProxyConfig proxyConfig = CustomTokenProxyConfiguration.parseAndValidate(configuration);
+            if (proxyConfig != null) {
+                identityClientOptions.setHttpClient(new CustomTokenProxyHttpClient(proxyConfig));
+            }
```

Comment:
The validation and http client creation part can be moved to the builder.
So, the constructor can remain exception throw free, for chained credentials to instantiate error free.